leetcode/iter_1.rs

```diff
@@ -0,0 +1,48 @@
+//Final Value of Variable
+impl Solution {
+    pub fn final_value_after_operations(operations: Vec<String>) -> i32 {
+        let mut x = 0;
+
+        for op in operations {
+            match op.as_str() {
+                "--X" | "X--" => x -= 1,
+                "++X" | "X++" => x += 1,
+                _ => (),
+            }
+        }
+
+        x
+    }
+}
+
+//Number of Steps
+impl Solution {
+    pub fn number_of_steps(mut num: i32) -> i32 {
+        let mut count = 0;
+
+        while num > 0 {
+            if num % 2 == 0 {
+                num /= 2;
+                count += 1;
```

Comment:
Line 26 and Line 29 are identical. When there's a common code in both blocks of `if`, it is better to move the common code out of `if` as follows:

```rust
if num % 2 == 0 {
  num /= 2;
} else {
  num -= 1;
}
count += 1;
```